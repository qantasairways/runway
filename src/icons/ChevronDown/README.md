```js
<ChevronDown width="50" height="50" />
```
