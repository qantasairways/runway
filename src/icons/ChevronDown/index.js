import React from 'react';
import PropTypes from 'prop-types';
import { generate } from 'shortid';

export default function ChevronDown({ width, height, className }) {
  const pathId = generate();
  const maskId = generate();

  return (
    <svg
      viewBox="0 0 24 24"
      width={width}
      height={height}
      className={className}
      focusable="false"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <defs>
        <path
          id={pathId}
          d="M2.12 1.29L6 5.17l3.88-3.88a.996.996 0 1 1 1.41 1.41L6.7 7.29a.996.996 0 0 1-1.41 0L.7 2.7a.996.996 0 0 1 0-1.41c.39-.38 1.03-.39 1.42 0z"
        />
      </defs>
      <g fillRule="evenodd" transform="translate(6 8)">
        <mask id={maskId}>
          <use xlinkHref={`#${pathId}`} />
        </mask>
        <use fillRule="nonzero" xlinkHref={`#${pathId}`} />
        <g mask={`url(#${maskId})`}>
          <path d="M0 0h12v8H0z" />
        </g>
      </g>
    </svg>
  );
}

ChevronDown.propTypes = {
  width: PropTypes.string,
  height: PropTypes.string,
  className: PropTypes.string
};

ChevronDown.defaultProps = {
  width: '24',
  height: '24',
  className: ''
};
