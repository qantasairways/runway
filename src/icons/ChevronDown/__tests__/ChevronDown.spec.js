import React from 'react';
import { shallow } from 'enzyme';
import shortid from 'shortid';

import ChevronDown from '..';

jest.mock('shortid');

describe('ChevronDown', () => {
  it('should render', () => {
    shortid.generate.mockReturnValue('1a2b3c');
    const component = shallow(
      <ChevronDown width="36px" height="36px" className="my-favourite-icon" />
    );

    expect(component).toMatchSnapshot();
  });

  it('should set the width', () => {
    const component = shallow(<ChevronDown width="36px" />);
    const result = component.find('svg').props().width;

    expect(result).toEqual('36px');
  });

  it('should set the height', () => {
    const component = shallow(<ChevronDown height="36px" />);
    const result = component.find('svg').props().height;

    expect(result).toEqual('36px');
  });

  it('should set the class name', () => {
    const component = shallow(<ChevronDown className="my-favourite-icon" />);
    const result = component.find('svg').props().className;

    expect(result).toEqual('my-favourite-icon');
  });
});
