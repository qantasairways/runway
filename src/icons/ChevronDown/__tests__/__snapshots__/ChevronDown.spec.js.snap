// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChevronDown should render 1`] = `
<svg
  className="my-favourite-icon"
  height="36px"
  viewBox="0 0 24 24"
  width="36px"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <defs>
    <path
      d="M2.12 1.29L6 5.17l3.88-3.88a.996.996 0 1 1 1.41 1.41L6.7 7.29a.996.996 0 0 1-1.41 0L.7 2.7a.996.996 0 0 1 0-1.41c.39-.38 1.03-.39 1.42 0z"
      id="1a2b3c"
    />
  </defs>
  <g
    fillRule="evenodd"
    transform="translate(6 8)"
  >
    <mask
      id="1a2b3c"
    >
      <use
        xlinkHref="#1a2b3c"
      />
    </mask>
    <use
      fillRule="nonzero"
      xlinkHref="#1a2b3c"
    />
    <g
      mask="url(#1a2b3c)"
    >
      <path
        d="M0 0h12v8H0z"
      />
    </g>
  </g>
</svg>
`;
